### ISSUE TITLE
websocket protocol always uses 1338 port instead of configuraton

### ISSUE THREAD

**Issue by dpap** [Agile-IoT/agile-sdk#28]
When using websockets agile-sdk doesn't use right port. Instead of using the port in api it uses the default value.

```
var agile = require('./dist/index.js')({
  api: 'http://localhost:8080' //,
//  idm: 'http://localhost:3000',
//  token: "LQCL7C14y84Ayqedjmbm1LuIes1TsSyn5Cv"
})
const deviceId = 'ble84EB1866ECE3';
const componentID = 'PULSE';
```

![wsbug](https://user-images.githubusercontent.com/2686270/29918588-9f674b06-8e4e-11e7-8243-4787a2417bb8.JPG)

**Comment by craigmulligan** [comment]
Thanks for reporting, looks like we are mutating the parsed url. 

https://github.com/Agile-IoT/agile-sdk/blob/master/src/index.js#L34-L36

Will PR a fix soon.

**Comment by craigmulligan** [comment]
Fixed with `v1.1.0`
